Remotion/Web/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriter.cs

```diff
@@ -0,0 +1,119 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+#if !NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Web.UI;
+using Remotion.Web.UI;
+
+namespace Remotion.Web.ContentSecurityPolicy
+{
+  public interface INonceGenerator
+  {
+    string GenerateAlphaNumericNonce ();
+  }
+
+  public class NonceGenerator : INonceGenerator
+  {
+    public string GenerateAlphaNumericNonce ()
+    {
+      Span<byte> data = stackalloc byte[8];
+      using (var randomNumberGenerator = RandomNumberGenerator.Create())
+      {
+        randomNumberGenerator.GetBytes(data);
+      }
+
+      return Convert.ToHexString(data);
+    }
+  }
+
+  public class CspEnabledHtmlTextWriter : HtmlTextWriter
+  {
+    private static readonly Dictionary<string, string> s_supportedEvents =
+        new(StringComparer.OrdinalIgnoreCase)
+        {
+            { "onclick", "click" },
+            { "onchange", "change" },
+            { "onmouseover", "mouseover" },
+            { "onmouseout", "mouseout" },
+            { "onkeyup", "keyup" },
+            { "onkeydown", "keydown" },
+            { "onkeypress", "keypress" }
+        };
+
+    private readonly List<Tuple<string, string>> _registeredEvents = new();
```

Comment:
we discussed usign a ValueTuple here. please update. while doing so, please introduce names for item1 and item2 as this is supported by c# on ValueTuple.. Makes for better readability.